### PR TITLE
add support for application/x-www-form-urlencoded bodies

### PR DESCRIPTION
**Pull request by ahl**
@plotnick  and @karencfv you might be interested in this, but no need to review if not. If I merge before you get to this I'll still attend to any review comments you might leave.

### CODE REVIEW

**Comment by alex** [comment]
First time today! :-)

File: progenitor-impl/src/method.rs [hunk 1]
```diff
@@ -1655,6 +1675,100 @@ impl Generator {
...
+
+        let content_type = BodyContentType::from_str(content_str)?;
+
+        let typ = match content_type {
+            BodyContentType::OctetStream => {
+                // For an octet stream, we expect a simple, specific schema:
+                // "schema": {
+                //     "type": "string",
+                //     "format": "binary"
+                // }
```
Comment (karencfv) [inline progenitor-impl/src/method.rs:1710]: Not sure if you meant to uncomment this code?

**Review by karencfv — APPROVED** [review]
Thanks for this! It'll be great to use the whole current nexus.json 🙇‍♀️

File: progenitor-impl/src/method.rs (hunk 1 above)
Comment (ahl) [reply progenitor-impl/src/method.rs:1710]: No that's the JSON schema

**Review by plotnick — COMMENTED** [review]
A little late to the party, but this looks good to me (though I admit to being unfamiliar with the progenitor code).

**Pull request merged**